nodes/geometry/intersect_line_line.py

```diff
@@ -0,0 +1,40 @@
+import bpy
+from ... base_types.node import AnimationNode
+
+class IntersectLineLineNode(bpy.types.Node, AnimationNode):
+    bl_idname = "an_IntersectLineLineNode"
+    bl_label = "Intersect Line Line"
+    bl_width_default = 160
+    searchTags = ["Closest Points on 2 Lines"]
+        
+    def create(self):
+        self.newInput("Vector", "Line 1 Start", "line1Start").value = [-2, 0, 0]
+        self.newInput("Vector", "Line 1 End", "line1End").value = [2, 0, 0]
+        self.newInput("Vector", "Line 2 Start", "line2Start").value = [1, -1, 0]
+        self.newInput("Vector", "Line 2 End", "line2End").value = [1, 1, 0]
+        
+        self.newOutput("Vector", "Intersection (Closest) 1", "int1")
+        self.newOutput("Vector", "Intersection (Closest) 2", "int2")
+        self.newOutput("Boolean", "Is Valid", "isValid")
```

Comment:
I prefer 'Intersect' instead of 'Is Valid'..

And "Closest Line 1" + "Closest Line 2" for the vector outputs. Also 'int#' is not a good name... When I hear 'int', I always think about integer. Just name them 'closest1', ..